application_sandbox/overlapping_frames/main.cpp

```diff
@@ -0,0 +1,649 @@
+        nullptr,                                   // pTexelBufferView
+    };
+
+    app.device()->vkUpdateDescriptorSets(app.device(), 1, &write, 0, nullptr);
+
+    return descriptor_set;
+}
+
+int main_entry(const entry::EntryData* data) {
+    data->logger()->LogInfo("Application Startup");
+
+    vulkan::VulkanApplication app(data->allocator(), data->logger(), data);
+
+    auto temp_images = buildTempImages(app, data);
+    vulkan::VulkanModel screen(data->allocator(), data->logger(), screen_data);
+
+    // Initialize Screen Model
+    auto init_cmd_buf = app.GetCommandBuffer();
+    app.BeginCommandBuffer(&init_cmd_buf);
+
+    screen.InitializeData(&app, &init_cmd_buf);
+    vulkan::VkFence init_fence = CreateFence(&app.device());
+    app.EndAndSubmitCommandBuffer(
+        &init_cmd_buf,
+        &app.render_queue(),
+        {},
+        {},
+        {},
+        init_fence.get_raw_object()
+    );
+    app.device()->vkWaitForFences(
+        app.device(),
+        1,
+        &init_fence.get_raw_object(),
+        VK_TRUE,
+        UINT64_MAX
+    );
+
+    // Default Sampler
+    auto sampler = CreateDefaultSampler(&app.device());
+
+    // Pipeline Layout
+    auto post_pipeline_layout = app.CreatePipelineLayout({{{
+        0,
+        VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER,
+        1,
+        VK_SHADER_STAGE_FRAGMENT_BIT,
+        nullptr
+    }}});
+
+    // Triangle Render Pass
+    auto render_pass_triangle = buildRenderPass(
+        app,
+        VK_IMAGE_LAYOUT_UNDEFINED,
+        VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL
+    );
+    auto pipeline_triangle = buildTrianglePipeline(app, &render_pass_triangle);
+    auto image_views_triangle = buildTempImageViews(app, temp_images, data);
+    auto framebuffers_triangle = buildFramebuffers(
+        app,
+        render_pass_triangle,
+        image_views_triangle,
+        data
+    );
+
+    // Post Render Pass
+    auto render_pass_post = buildRenderPass(
+        app,
+        VK_IMAGE_LAYOUT_UNDEFINED,
+        VK_IMAGE_LAYOUT_PRESENT_SRC_KHR
+    );
+    auto pipeline_post = buildPostPipeline(app, post_pipeline_layout, &render_pass_post, &screen);
+    auto image_views_post = buildSwapchainImageViews(app, data);
+    auto framebuffers_post = buildFramebuffers(
+        app,
+        render_pass_post,
+        image_views_post,
+        data
+    );
+
+    // Triangle Synchronization
+    containers::vector<vulkan::VkCommandBuffer> command_buffers_triangle(data->allocator());
+    containers::vector<vulkan::VkSemaphore> render_finished_triangle(data->allocator());
+
+    // Post Synchronization
+    containers::vector<vulkan::VkSemaphore> image_acquired(data->allocator());
+    containers::vector<vulkan::VkSemaphore> render_finished_post(data->allocator());
+    containers::vector<CommandTracker> command_trackers_post(data->allocator());
+    containers::unordered_map<uint32_t, uint32_t> progress_post(data->allocator());
+
```

Comment:
We should reserve the space for all values in the vector (saves re-allocation).

Reply:
Done